homeassistant/components/discovergy/__init__.py

```diff
@@ -20,35 +17,21 @@
 PLATFORMS = [Platform.SENSOR]
 
 
-@dataclass
-class DiscovergyData:
-    """Discovergy data class to share meters and api client."""
-
-    api_client: Discovergy
-    meters: list[Meter]
-    coordinators: dict[str, DiscovergyUpdateCoordinator]
-
-
 async def async_setup_entry(hass: HomeAssistant, entry: ConfigEntry) -> bool:
     """Set up Discovergy from a config entry."""
-    hass.data.setdefault(DOMAIN, {})
-
-    # init discovergy data class
-    discovergy_data = DiscovergyData(
-        api_client=Discovergy(
-            email=entry.data[CONF_EMAIL],
-            password=entry.data[CONF_PASSWORD],
-            httpx_client=get_async_client(hass),
-            authentication=BasicAuth(),
-        ),
-        meters=[],
-        coordinators={},
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = []
```

Comment:
What do you think about storing the list of coordinators in a variable `coordinators` before assigning? Makes it a bit better to read, but this is just an opinion, feel free to ignore

Reply:
Will do, is indeed better to read